src/api/CoinBase.ts

```diff
@@ -0,0 +1,63 @@
+import { ExchangeApi } from '../types/ExchangeApi';
+import fetch, { Response } from 'node-fetch';
+import BigNumber from 'bignumber.js';
+
+const URL = 'https://api.coinbase.com/v2/exchange-rates';
+
+export type CoinBaseResponse = {
+    data: {
+        currency: string;
+        rates: Record<string, string>;
+    };
+};
+
+const CURRENCY_MAPPING: Record<string, string> = {
+    RIF: 'RIF',
+    RBTC: 'RBTC',
+    TKN: 'RIF'
+};
+
+export class CoinBase implements ExchangeApi {
+    getApiTokenName(tokenSymbol: string): string {
+        if (!tokenSymbol) {
+            throw Error('ExchangeApi cannot map a token with a null value');
+        }
+        const resultMapping = CURRENCY_MAPPING[tokenSymbol];
+        if (resultMapping) {
+            return resultMapping;
+        }
+        return tokenSymbol;
+    }
```

Comment:
If the tokenSymbol is not in the CURRENCY_MAPPING, this function returns the tokenSymbol itself. How can a caller distinguish between the case when the tokenSymbol is not in the CURRENCY_MAPPING and the case when the API name is the same as the tokenSymbol?

Reply:
I don't think that the caller should care about this. If the caller needs his token to be in the mapping, he will need to add it manually.

Reply:
updated